contracts/NonfungiblePositionManager.sol

```diff
@@ -0,0 +1,142 @@
+    //     require(params.amount1Min <= uint256(uint128(delta.amount1())), "INSUFFICIENT_AMOUNT1");
+    // }
+
+    function increaseLiquidity(uint256 tokenId, uint256 liquidity, bytes calldata hookData, bool claims)
+        external
+        isAuthorizedForToken(tokenId)
+        returns (BalanceDelta delta)
+    {
+        delta = _increaseLiquidity(tokenPositions[tokenId].range, liquidity, hookData, claims, msg.sender);
+    }
+
+    function decreaseLiquidity(uint256 tokenId, uint256 liquidity, bytes calldata hookData, bool claims)
+        public
+        isAuthorizedForToken(tokenId)
+        returns (BalanceDelta delta)
+    {
+        delta = _decreaseLiquidity(tokenPositions[tokenId].range, liquidity, hookData, claims, msg.sender);
+    }
+
+    function burn(uint256 tokenId, address recipient, bytes calldata hookData, bool claims)
+        external
+        isAuthorizedForToken(tokenId)
+        returns (BalanceDelta delta)
+    {
+        // remove liquidity
+        TokenPosition storage tokenPosition = tokenPositions[tokenId];
+        LiquidityRangeId rangeId = tokenPosition.range.toId();
+        Position storage position = positions[msg.sender][rangeId];
+        if (0 < position.liquidity) {
+            decreaseLiquidity(tokenId, position.liquidity, hookData, claims);
+        }
+        require(position.tokensOwed0 == 0 && position.tokensOwed1 == 0, "NOT_EMPTY");
+        delete positions[msg.sender][rangeId];
+        delete tokenPositions[tokenId];
+
+        // burn the token
+        _burn(tokenId);
+    }
+
+    // TODO: in v3, we can partially collect fees, but what was the usecase here?
+    function collect(uint256 tokenId, address recipient, bytes calldata hookData, bool claims)
+        external
+        returns (BalanceDelta delta)
+    {
+        delta = _collect(tokenPositions[tokenId].range, hookData, claims, msg.sender);
+    }
+
+    function feesOwed(uint256 tokenId) external view returns (uint256 token0Owed, uint256 token1Owed) {
+        TokenPosition memory tokenPosition = tokenPositions[tokenId];
+        return feesOwed(tokenPosition.owner, tokenPosition.range);
+    }
+
+    function _afterTokenTransfer(address from, address to, uint256 firstTokenId, uint256 batchSize) internal override {
+        TokenPosition storage tokenPosition = tokenPositions[firstTokenId];
+        LiquidityRangeId rangeId = tokenPosition.range.toId();
+        Position storage position = positions[from][rangeId];
+        position.operator = address(0x0);
+
+        // transfer position data to destination
+        positions[to][rangeId] = position;
+        delete positions[from][rangeId];
+
+        // update token position
+        tokenPositions[firstTokenId] = TokenPosition({owner: to, range: tokenPosition.range});
+    }
+
+    modifier isAuthorizedForToken(uint256 tokenId) {
+        require(_isApprovedOrOwner(msg.sender, tokenId), "Not approved");
```

Comment:
hm I think _isApprovedOrOwner actually got deprecated in the most up to date version of the OZ ERC721 contracts.

solmates version also doesnt use this

so we should look into updating? or is there a specific reason to still use this?

Reply:
can look into updating -- grabbed it from v3